### PR TITLE
Alerting: deactivate an Alertmanager configuration

### PR DESCRIPTION
**Pull request by papagian**
**What this PR does / why we need it**:
It implements DELETE `/api/alertmanager/grafana/config/api/v1/alerts` by storing the default configuration which stops the existing configuration from being in use.

**Which issue(s) this PR fixes**:

Fixes grafana/alerting-squad#171

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/services/ngalert/api/api_alertmanager.go [hunk 1]
```diff
@@ -45,8 +45,15 @@ func (srv AlertmanagerSrv) RouteCreateSilence(c *models.ReqContext, postableSile
...
 
 func (srv AlertmanagerSrv) RouteDeleteAlertingConfig(c *models.ReqContext) response.Response {
-	// not implemented
-	return NotImplementedResp
+	if !c.HasUserRole(models.ROLE_EDITOR) {
+		return ErrResp(http.StatusForbidden, errors.New("permission denied"), "")
+	}
+	if err := srv.am.SaveAndApplyDefaultConfig(); err != nil {
+		srv.log.Error("unable to save and apply default alertmanager configuration", "err", err)
+		return ErrResp(http.StatusBadRequest, err, "failed to save and apply default Alertmanager configuration")
```
Comment (davidmparrott) [inline pkg/services/ngalert/api/api_alertmanager.go:53]: nit: does a 400 or a 500 make more sense here? Could be a reason other than a bad request that we failed to save the config

**Review by davidmparrott — APPROVED** [review]
Nice solution of applying the default config when DELETE is called

File: pkg/services/ngalert/api/api_alertmanager.go (hunk 1 above)
Comment (papagian) [reply pkg/services/ngalert/api/api_alertmanager.go:53]: I will change it; `BadRequest` does not make any sense in this case.

**Comment by papagian** [comment]
> Nice solution of applying the default config when DELETE is called

credit to @gotjosh who suggested so!

**Pull request merged**